src/mod/thr/thr.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * based on climb_gps.c
+ *
+ *  Created on: 27.02.2022
+ *      Copy paste by: Jevgeni
+ */
+#include "thr.h"
+#include <ado_uart.h>
+#include "../l7_climb_app.h"
+#include "../l2_debug_com.h"
+#include "../l4_thruster.h" // with that we include variable  l4_thr_ExpectedReceiveBuffer  which defines expected RX buffer length
+
+
+// prototypes
+void thrUartIRQ(LPC_USART_T *pUART);
+void thrProcessRxByte(uint8_t rxByte);
+void thr_debugPrintBuffer(uint8_t *buffer,int bufferlen);
+
+static thr_initdata_t *thrInitData;
+
+// ************************** TX Circular byte Buffer Begin ********************
+#define THR_TX_BUFFERSIZE	200
+static uint8_t 		   thrTxBuffer[THR_TX_BUFFERSIZE];
+static uint8_t         thrTxWriteIdx = 0;
+static uint8_t         thrTxReadIdx  = 0;
+static bool		  	   thrTxBufferFull = false;
+static bool				thrFirstByteAfterReset=true;
+
+
+
+#define THR_BUFFERLEN 700 // maximum buffer length
+//int thr_counter = 0; would be defined in l4_thruster
+char thr_receiveBuffer[THR_BUFFERLEN] ="";
+//l4_thr_counter = 0; // set received bytes counter to 0 initialy // defined in l4_thruster.c
+
+
+static bool inline thrTxBufferEmpty() {
+	if (thrTxBufferFull) {
+		return false;
+	} else {
+		return (thrTxReadIdx == thrTxWriteIdx);
+	}
+}
+
+static bool inline thrTxAddByte(uint8_t b) {
+	if (!thrTxBufferFull) {
+		thrTxBuffer[thrTxWriteIdx++] = b;
+		if (thrTxWriteIdx >= THR_TX_BUFFERSIZE) {
+			thrTxWriteIdx = 0;
+		}
+		if (thrTxWriteIdx == thrTxReadIdx) {
+			thrTxBufferFull = true;
+		}
+		return true;
+	} else {
+		return false;
+	}
+}
+
+static uint8_t inline thrTxGetByte(void) {
+	uint8_t retVal = thrTxBuffer[thrTxReadIdx++];
+	if (thrTxReadIdx >= THR_TX_BUFFERSIZE) {
+		thrTxReadIdx = 0;
+	}
+	thrTxBufferFull = false;
+	return retVal;
+}
+
+// ************************** TX Circular byte Buffer End ********************
+
+
+void thrInit (void *initData) {
+	thrInitData = (thr_initdata_t*) initData;
+
+	// Switch the 'enable' pin to low (no internal Volage regulator needed
+	//Chip_GPIO_SetPinOutLow(LPC_GPIO, thrInitData->pEnablePin->pingrp, thrInitData->pEnablePin->pinnum);
+	//// IMPORTANT PINIDX_RS485_TX_RX   HIGH - TRANSMIT              LOW - RECEIVE
+	// SET TO RECEIVE UPON INITIALIZATION
+	//Chip_GPIO_SetPinOutLow(LPC_GPIO, 2, 5); //  This is PINIDX_RS485_TX_RX RECEIVE
+	//Chip_GPIO_SetPinOutHigh(LPC_GPIO, 2, 5); //  This is PINIDX_RS485_TX_RX TRANSMIT
+
+	// Init UART
+	InitUart(thrInitData->pUart, 115200, thrUartIRQ);
+	thrFirstByteAfterReset = true;
+
+	// Enable dirrection controll
+	LPC_UART1->RS485CTRL |= UART_RS485CTRL_DCTRL_EN; // Enable Auto Direction Control
```

Comment:
here and next lines  you should use the configured uart and not a hardcoded one !

Reply:
https://github.com/RobertK66/obc_1769_core/pull/34/commits/34ed796648ff98ced677495b560fa473a2707849

Fixed